tools/simpleui/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/gorilla/mux"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+	"google.golang.org/grpc"
+)
+
+var (
+	resultTmpl = template.Must(template.New("results.html").ParseFiles("templates/results.html"))
+	recordTmpl = template.Must(template.New("records.html").
+			Funcs(template.FuncMap{
+			"textproto": textproto,
+		}).
+		ParseFiles("templates/records.html"))
+)
+
+func (u *ui) home(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprint(w, "Add namespace to URL (/<namespace>) ^^^")
+}
+
+func (u *ui) results(w http.ResponseWriter, r *http.Request) {
+	res, err := u.client.ListResults(r.Context(), &pb.ListResultsRequest{
+		Parent: strings.Trim(r.URL.Path, "/"),
+	})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintln(w, err)
+		return
+	}
+	if err := resultTmpl.Execute(w, res); err != nil {
+		return
+	}
+}
+
+func (u *ui) records(w http.ResponseWriter, r *http.Request) {
+	res, err := u.client.ListRecords(r.Context(), &pb.ListRecordsRequest{
+		Parent: strings.Trim(r.URL.Path, "/"),
+	})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintln(w, err)
+		return
+	}
+	if err := recordTmpl.Execute(w, res); err != nil {
+		return
+	}
+}
+
+type ui struct {
+	client pb.ResultsClient
+}
+
+func main() {
+	conn, err := grpc.Dial("localhost:50051", grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+	defer conn.Close()
+	u := &ui{
+		client: pb.NewResultsClient(conn),
+	}
+
+	r := mux.NewRouter()
+	r.HandleFunc("/", u.home)
+	r.HandleFunc("/{namespace}", u.results)
+	r.HandleFunc("/{namespace}/results/{name}", u.records)
+	http.Handle("/", r)
+
+	hostname, _ := os.Hostname()
+	log.Printf("Running on http://%s:8080\n", hostname)
```

Comment:
Tiny error here: the hostname doesn't work for me: 

> 2021/02/01 16:12:11 Running on http://xinruzhang-macbookpro.roam.corp.google.com:8080

Using `localhost` instead solved the problem. Is there any other function can fit this requirement?

Reply:
Done.